packages/underscore-redirects/src/astro.ts

```diff
@@ -0,0 +1,123 @@
+import type { AstroConfig, RouteData, ValidRedirectStatus } from 'astro';
+import { Redirects } from './redirects.js';
+import { posix } from 'node:path';
+
+const pathJoin = posix.join;
+
+function getRedirectStatus(route: RouteData): ValidRedirectStatus {
+	if(typeof route.redirect === 'object') {
+		return route.redirect.status;
+	}
+	return 301;
+}
+
+export function createRedirectsFromAstroRoutes(
```

Comment:
I find it difficult to understand the logic of this function:
- the PR doesn't explain much about the shared code
- there aren't any comments that help

If I would have to fix this code in the future, I would have hard time 😅